packages/web-react/src/components/Tooltip/useFloatingUI.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Placement,
+  arrow,
+  autoUpdate,
+  flip,
+  inline,
+  limitShift,
+  offset,
+  shift,
+  size,
+  useClick,
+  useFloating,
+  useInteractions,
+  useRole,
+} from '@floating-ui/react';
+import { useState } from 'react';
+
+type UseTooltipUIProps = {
+  arrowRef: React.MutableRefObject<HTMLElement | null>;
+  flipProp: boolean;
+  flipCrossAxis: boolean;
+  flipFallbackAxisSideDirection: 'none' | 'start' | 'end';
+  flipFallbackPlacements?: Placement | Placement[];
+  isOpen?: boolean;
+  onToggle: (isOpen: boolean) => void;
+  tooltipPlacement?: Placement;
+  shiftProp: boolean;
+  sizeProp: boolean;
+  offset?: number;
+  tooltipMaxWidth?: number;
+};
+
+// Convert a string or array of strings to an array of strings (for placements)
+const stringToArray = (value: Placement | Placement[]): Placement[] | undefined =>
+  Array.isArray(value) ? value : [value];
+
+export const useFloatingUI = (props: UseTooltipUIProps) => {
```

Comment:
This is not blocking but maybe it should be better not to stick with the implementation. So I suggest renaming this to just `useFloating` so we do not care in the future if internal implementation uses this or that library. What do you think?

Reply:
renamed as you suggested